server/src/providers/regression-model.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+const MLR = require('ml-regression-multivariate-linear');
+import * as fs from 'fs';
+@Injectable()
+export class RegressionModel {
+    private model: any;
+    private logsEnabled: boolean;
+    private inputs: any;
+    private outputs: any;
+    constructor(logsEnabled= false) {
+        this.logsEnabled = logsEnabled;
+        this.log('Logs are enabled');
+    }
+    // Trains the model with the inputs
+    public trainModel(inputs, output) {
+        try {
+            this.model = new MLR(inputs, output);
+            this.log('Model has been trained');
+        } catch (err){
+            throw err;
+        }
+    }
+
+    // Prediction of the model
+    public predict(inputs) {
+        try {
+            const predictions = [];
+            inputs.forEach(input => {
+                predictions.push(this.model.predict(input));
+            });
+            this.log('Model has predicted');
+            return predictions;
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    // Saves a model to HDD 
+    public saveModel(modelName) {
+        try {
+        const jsonModel = JSON.stringify(this.model.toJSON());
+        fs.writeFileSync('ai_models/' + modelName + '.json', jsonModel);
```

Comment:
```suggestion
        fs.writeFileSync(`'ai_models/${modelName}.json`, jsonModel);
```

Maybe abstract 'ai_models' into a constant at the top of the file so it can be changed easily.